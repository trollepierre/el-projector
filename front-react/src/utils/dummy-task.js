const dummyTask = (task) => ({
  id: 1,
  name: 'Build el-projector',
  description: 'more data about how to do it',
  points: 999, // given by server
  createdDate: '19/05/2019',
  endDate: undefined, // must be done before this date
  optimalDate: undefined, // user would do it at this date
  isDone: false,
  doneDate: undefined, // for log record
  priority: 'HIGHLY_IMPORTANT', // Enum with 5 levels (inherited by default, but can be updated)
  benefit: 'ORGASM_HAPPINESS', // Enum with 5 levels
  neededTime: 'ONE_MONTH', // Enum
  epicId: 'A1', // Yearly happiness epic
  neededMaterial: ['MacTo', 'iMoi', 'MacNo', 'home'],
  subTaskIds: ['TASK10', 'TASK11', 'TASK12'], // checklist of subtasks that depends on it
  masterTaskId: 'TASK0', // this task depends on the master one
  ...task,
})

export { dummyTask }
