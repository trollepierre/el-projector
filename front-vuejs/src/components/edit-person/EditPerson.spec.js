import EditTask from './EditTask.vue'
import { dummyTask } from '../../utils/test/dummy-task'

describe('EditTask.vue', () => {
  it('should match snapshot', () => {
    // When
    const wrapper = shallowMount(EditTask, { propsData: { task: dummyTask() } })

    // Then
    expect(wrapper.element).toMatchSnapshot()
  })
})
