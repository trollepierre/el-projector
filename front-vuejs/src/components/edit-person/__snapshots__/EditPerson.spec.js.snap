// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTask.vue should match snapshot 1`] = `
<tr>
  <td
    class="next"
  >
    <input />
  </td>

  <td
    class="first-name"
  >
    <input />
  </td>

  <td
    class="last-name"
  >
    <input />
  </td>

  <td
    class="points"
  >
    <input />
  </td>

  <td
    class="adjusted-points"
  >
    <input />
  </td>

  <td
    class="last-date"
  >
    <input />
  </td>

  <td
    class="impact"
  />

  <td
    class="birth-date"
  >
    <input />
  </td>

  <td
    class="origin"
  >
    <input />
  </td>

  <td>
    <appbutton-stub
      backgroundcolor="blue"
      color="white"
      text="save"
    />

    <appbutton-stub
      backgroundcolor=""
      color=""
      text="delete"
    />
  </td>
</tr>
`;
